### ISSUE TITLE
Data loading speed is too slow

### ISSUE THREAD

**Issue by Minju-Jung** [kenshohara/3D-ResNets-PyTorch#48]
Thank you for providing a nice code!

I tested the pretrained model "resnext-101-64f-kinetics-ucf101_split1.pth" using UCF 101.
I got 93.99% video level accuracy.

However, the computational speed is really slow (roughly a few hours) because of data loading.
Although I'm using HDD not SSD, data loading speed is much slower than my expectation.
Especially, (4n+1) times speed is significantly slow (4 is the number of threads).
I added my log file as follows:

[1/197]	Time 335.936 (335.936)	Data 333.094 (333.094)	
[2/197]	Time 1.754 (168.845)	Data 0.000 (166.547)	
[3/197]	Time 1.758 (113.149)	Data 0.000 (111.032)	
[4/197]	Time 1.769 (85.304)	Data 0.000 (83.274)	
[5/197]	Time 298.968 (128.037)	Data 297.199 (126.059)	
[6/197]	Time 1.750 (106.989)	Data 0.000 (105.049)	
[7/197]	Time 1.760 (91.956)	Data 0.000 (90.042)	
[8/197]	Time 1.757 (80.681)	Data 0.000 (78.787)	
[9/197]	Time 280.848 (102.922)	Data 279.067 (101.040)	
[10/197]	Time 1.766 (92.807)	Data 0.000 (90.936)	
[11/197]	Time 1.754 (84.529)	Data 0.000 (82.669)	
[12/197]	Time 1.760 (77.632)	Data 0.000 (75.780)	
[13/197]	Time 290.565 (94.011)	Data 288.792 (92.166)	
[14/197]	Time 1.750 (87.421)	Data 0.000 (85.582)	
[15/197]	Time 1.763 (81.711)	Data 0.000 (79.877)	
[16/197]	Time 1.756 (76.713)	Data 0.000 (74.885)	
[17/197]	Time 303.138 (90.032)	Data 301.384 (88.208)	
[18/197]	Time 4.009 (85.253)	Data 2.253 (83.433)	
[19/197]	Time 3.780 (80.965)	Data 2.027 (79.148)	
[20/197]	Time 1.759 (77.005)	Data 0.000 (75.191) ...	

Is the speed is normal or something wrong?
If it is something wrong, could you kindly let me know how to fix it?

**Comment by kenshohara** [comment]
In my environment, the loading speed is 100x faster than your results, though I used SSD...
But I'm not sure why your loading is such slow.
For the data loading, I just use the data loader of pytorch.

**Comment by Minju-Jung** [comment]
Thank you for prompt reply!

Based on your reply, the problem comes from my side.
I will check my simulation speed using SSD.

If you don't mind, could you explain why my accuracy 93.99% is sightly different from the reported accuracy 94.5% in the paper?

**Comment by kenshohara** [comment]
I reported the accuracy averaged over 3 splits in the paper.
The accuracy of split1 evaluated in my environment is the same as your 93.99%.

**Comment by Minju-Jung** [comment]
Now I understand the accuracy difference!
Thank you for kind reply.

**Comment by kenshohara** [comment]
I was mistaken.
When using sample_duration=64,
data loading for each batch is slower than when using sample_duration=16.
This is the results in my environment.

[1/197]	Time 22.136 (22.136)	Data 8.699 (8.699)	
[2/197]	Time 0.500 (11.318)	Data 0.000 (4.349)	
[3/197]	Time 0.507 (7.714)	Data 0.000 (2.900)	
[4/197]	Time 0.499 (5.910)	Data 0.000 (2.175)	
[5/197]	Time 0.503 (4.829)	Data 0.000 (1.740)	
[6/197]	Time 0.505 (4.108)	Data 0.000 (1.450)	
[7/197]	Time 0.501 (3.593)	Data 0.000 (1.243)	
[8/197]	Time 0.499 (3.206)	Data 0.000 (1.088)	
[9/197]	Time 0.504 (2.906)	Data 0.000 (0.967)	
[10/197]	Time 5.464 (3.162)	Data 4.934 (1.363)	
[11/197]	Time 0.618 (2.931)	Data 0.065 (1.245)	
[12/197]	Time 0.513 (2.729)	Data 0.000 (1.142)	
[13/197]	Time 0.515 (2.559)	Data 0.000 (1.054)	
[14/197]	Time 7.007 (2.877)	Data 6.491 (1.442)	
[15/197]	Time 0.514 (2.719)	Data 0.000 (1.346)	
[16/197]	Time 0.598 (2.586)	Data 0.076 (1.267)	
[17/197]	Time 0.506 (2.464)	Data 0.000 (1.192)	
[18/197]	Time 6.776 (2.704)	Data 6.266 (1.474)	
[19/197]	Time 0.787 (2.603)	Data 0.285 (1.412)	
[20/197]	Time 0.992 (2.522)	Data 0.491 (1.365)	
[21/197]	Time 0.514 (2.427)	Data 0.000 (1.300)	
[22/197]	Time 6.376 (2.606)	Data 5.843 (1.507)	
[23/197]	Time 0.576 (2.518)	Data 0.070 (1.444)	
[24/197]	Time 0.721 (2.443)	Data 0.207 (1.393)	
[25/197]	Time 0.510 (2.366)	Data 0.000 (1.337)

**Comment by vateye** [comment]
@Minju-Jung How do you evaluate on the test set? Would you like share your running and compute accuracy script?

**Comment by Minju-Jung** [comment]
@kenshohara Thank you for your information.
I tested the same experiment using SSD, and then the speed problem is solved!
I think that my HDD is something wrong...
